### ISSUE TITLE
Use Gcloud default authentication instead of an API key

### ISSUE THREAD

**Issue by MarkEdmondson1234** [promptfoo/promptfoo#787]
Hi, thanks for the great package.  I am self hosting on Cloud Run and attempting to hook it up to the self-hosted Langfuse too.

I am using the new bucket mount feature to write the Prompt foo evaluations to disk, and calling the promptfoo eval within cloudbuild.yaml.

When running within GCP, there is an authentication token available in the metadata, so I don't need to use API keys etc.  It would be great if the vertex ai check could reuse this authentication, which is assigned at the service key level without needing the less secure JSON key.

e.g. currently:

```
Error: Google Vertex API key is not set. Set the VERTEX_API_KEY environment variable or add `apiKey` to the provider config. You can get an API token by running `gcloud auth print-access-token`

Error: Google Vertex API key is not set. Set the VERTEX_API_KEY environment variable or add `apiKey` to the provider config. You can get an API token by running `gcloud auth print-access-token`
    at VertexChatProvider.callApi (/usr/local/lib/node_modules/promptfoo/dist/src/providers/vertex.js:105:19)
    at Evaluator.runEval (/usr/local/lib/node_modules/promptfoo/dist/src/evaluator.js:277:62)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async processEvalStep (/usr/local/lib/node_modules/promptfoo/dist/src/evaluator.js:594:25)
 ```   
 
I don't know any node to offer a PR, but I'm sure there is a line that can be added that checks for default authentication details before looking for an API key.

**Comment by typpo** [comment]
Thanks for flagging this, Mark.  We definitely should get it working with gcloud's default application credentials.  It looks like the code already supports it, but there might be an extra apiKey check that we don't need.

When you're running locally, are you using default application credentials or manually providing the API key via `gcloud auth print-access-token`?

**Comment by MarkEdmondson1234** [comment]
I’ve not actually run locally yet - only via cloud build.

**Comment by typpo** [comment]
Got it.  Will have a look and include the fix in the next release, so that it picks up [default application credentials](https://cloud.google.com/docs/authentication/application-default-credentials).

**Comment by MarkEdmondson1234** [comment]
Great thanks!

**Comment by typpo** [comment]
Fix landed in 0.59.0!  LMK how it works

**Comment by MarkEdmondson1234** [comment]
Awesome, will try it out
